File: app/src/test/java/com/example/megaport/mynews/ArticlesTest.java
package com.example.megaport.mynews;

import com.example.megaport.mynews.Models.Article;
import com.example.megaport.mynews.Models.Articles;
import org.junit.Test;
import java.util.ArrayList;
import java.util.List;
import static org.junit.Assert.assertEquals;

public class ArticlesTest {

    private final Article article = new Article();
    private final Articles articles = new Articles();

    @Test
    public void setArticles() {
        String testStr = "test";
        article.setTitle(testStr);
        List<Article> articlesL = new ArrayList<>();
        articlesL.add(article);
        articles.setArticles(articlesL);
        List<Article> articles2 = articles.getArticles();
        assertEquals("Articles getter or setter is invalid", testStr, articles.getArticle(0).getTitle());
        assertEquals("Articles getter or setter is invalid", testStr, articles2.get(0).getTitle());
    }
}

File: app/src/test/java/com/example/megaport/mynews/ArticlesTest.kt
package com.example.megaport.mynews

import com.example.megaport.mynews.models.Article
import com.example.megaport.mynews.models.Articles
import org.junit.Test
import java.util.ArrayList
import java.util.Objects

import org.junit.Assert.assertEquals

class ArticlesTest {

    private val article = Article()
    private val articles = Articles()

    @Test
    fun setArticles() {
        val testStr = "test"
        article.title = testStr
        val articlesL = ArrayList<Article>()
        articlesL.add(article)
        articles.articles = articlesL
        val articles2 = articles.articles
        assertEquals("Articles getter or setter is invalid", testStr, Objects.requireNonNull<Article>(articles.getArticle(0)).title)
        assertEquals("Articles getter or setter is invalid", testStr, Objects.requireNonNull<List<Article>>(articles2)[0].title)
    }
}